src/TraceEvent/TraceLoggingEventId.cs

```diff
@@ -38,8 +40,8 @@ public void Dispose()
             if (m_traceLoggingEventMap == null)
                 return;
 
-            foreach (var key in m_traceLoggingEventMap.Keys)
-                Marshal.FreeHGlobal((IntPtr)key.Provider);
+            foreach (var kvp in m_traceLoggingEventMap)
+                Marshal.FreeHGlobal((IntPtr)kvp.Key.Provider);
```

Comment:
@pharring It depends on where things are running. It's already added to coreclr in dotnet/coreclr#8560 (not sure what release version that corresponds to). For Roslyn I added the `Deconstruct` extension method when I used the feature in dotnet/roslyn#19126.